### PR TITLE
feat: Multicall v2

### PR DESCRIPTION
**Pull request by hachiojidev**

### CODE REVIEW

File: src/config/constants/contracts.ts [hunk 1]
```diff
@@ -20,8 +20,8 @@ export default {
     56: '',
   },
   multiCall: {
-    56: '0x1ee38d535d541c55c9dae27b12edf090c608e6fb',
-    97: '0x67ADCB4dF3931b0C5Da724058ADC2174a9844412',
+    56: '0xfF6FD90A470Aaa0c1B8A54681746b07AcdFedc9B',
+    97: '',
```
Comment (hachiojidev) [inline src/config/constants/contracts.ts:24]: Removed testnet as I could not find a Multcall2 on there.

File: src/utils/multicall.ts [hunk 2]
```diff
@@ -11,15 +11,46 @@ interface Call {
...
-  const { returnData } = await multi.methods.aggregate(calldata).call()
-  const res = returnData.map((call, i) => itf.decodeFunctionResult(calls[i].name, call))
+    const calldata = calls.map((call) => [call.address.toLowerCase(), itf.encodeFunctionData(call.name, call.params)])
+    const { returnData } = await multi.methods.aggregate(calldata).call()
+    const res = returnData.map((call, i) => itf.decodeFunctionResult(calls[i].name, call))
 
-  return res
+    return res
+  } catch (error) {
+    throw new Error(error)
```
Comment (hachiojidev) [inline src/utils/multicall.ts:25]: Throwing the error here results in better console messages.

Comment (RabbitDoge) [reply src/utils/multicall.ts:25]: there is no reason to try catch if you re throw the error.

File: src/utils/multicall.ts [hunk 3]
```diff
@@ -11,15 +11,46 @@ interface Call {
...
+      const [result, data] = call
+      return {
+        result,
+        data: itf.decodeFunctionResult(calls[i].name, data),
+      }
+    })
+
+    return res
+  } catch (error) {
+    throw new Error(error)
```
Comment (RabbitDoge) [inline src/utils/multicall.ts:53]: same here

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**